src/lib/Imp.hs

```diff
@@ -773,14 +763,14 @@ splitDest (maybeDest, (Block decls ans)) = do
         | fmap (const ()) items == fmap (const ()) items' -> do
             zipWithM_ gatherVarDests (toList items) (toList items')
       (Con (ConRef (SumAsProd _ _ _)), _) -> tell [(dest, result)]  -- TODO
+      (_, ProjectElt _ _) -> tell [(dest, result)]  -- TODO: is this reasonable?
```

Comment:
It's quite possible that this is just totally incorrect. It seemed like everything was working fine without any special lowering step for `ProjectElt`, possibly because we automatically rewrite-away `ProjectElt` atoms anytime values are substituted? I don't really understand how the lowering step works, though. (I know for a fact that a test was failing without this line, but adding this line made the test pass again. But I don't know *why* this fixed the test, it might be a bug that just happened to work fine in the case that was being tested.)

Reply:
I thought about it a bit more and I think that the implementation you have here is correct. _However_, I would still like to better understand why was this necessary. I have a feeling that most `ProjectElt`s should be eliminated in simplification (at least those that would make their way to `Imp`). Could you please try to isolate the program where this happens? I might try to do that later myself...

Reply:
The simplest case I could come up with that breaks without this line is:
```
%passes simp imp
:p snd $ withState 42 \s. ()
> 42
> === simp ===
> Module (Simp)
>   unevaluated decls:
>   tmp:(Unit & Int32) =
>     %runState 42 (\h':Type. \_:(Ref h' Int32) ->{State h'} ())
>
>   evaluated bindings:
>   (_ans_ @> (Int32, ((\(_, b). b) tmp)))
> === imp ===
> def entryFun EntryFun CUDANotRequired []
>   v:(Ptr (a, CPU, Int32)) = alloc Int32[1]
>   v1:(Ptr (a, stack, Int32)) = alloc Int32[1]
>   store v1 42
>   v2:Int32 = load v1
>   store v v2
>   v3:Int32 = load v
>   return [v3]
```
Also, basically every test that we have starts failing without this, so it's not a rare situation, and probably gets triggered by other primitive operations that return tuples.

So far I just have a `TODO` here. If you have a justification for why what I've done is correct, I'd be happy to replace it with a comment describing why this works and why it is necessary.